### PR TITLE
UPPSF-488 Endpoint for annotation deletion

### PR DESCRIPTION
**Pull request by karsov**

### CODE REVIEW

File: mapper/mapper_test.go [hunk 1]
```diff
@@ -78,3 +78,19 @@ func TestDefaultPassThrough(t *testing.T) {
 
 	assert.JSONEq(t, string(expectedBody), string(actualBody), "json not equal")
 }
+
+func TestImplicitAnnotationsAreFiltered(t *testing.T) {
+
+	originalBody, err := ioutil.ReadFile("test_resources/implicit_annotations_v2.json")
+	if err != nil {
+		panic(err)
```
Comment (mchompalova) [inline mapper/mapper_test.go]: Should we panic here?

File: mapper/mapper_test.go [hunk 2]
```diff
@@ -78,3 +78,19 @@ func TestDefaultPassThrough(t *testing.T) {
...
+	}
+	expectedBody, err := ioutil.ReadFile("test_resources/implicit_annotations_PAC.json")
+
+	if err != nil {
+		panic(err)
```
Comment (mchompalova) [inline mapper/mapper_test.go]: And here.

File: mapper/mapper.go [hunk 3]
```diff
@@ -24,7 +26,8 @@ func ConvertPredicates(body []byte) ([]byte, error) {
 	convertedAnnotations := make([]map[string]interface{}, 0)
 	err := json.Unmarshal(body, &originalAnnotations)
 	if err != nil {
-		log.Fatal("Could not unmarshall json body", err)
+		log.WithError(err).WithField("body", string(body)).Error("Could not unmarshall json body")
```
Comment (mchompalova) [inline mapper/mapper.go]: unmarshal

File: mapper/mapper_test.go (hunk 1 above)
Comment (karsov) [reply mapper/mapper_test.go]: There is no "panic" in the code, not sure what you looked at.

File: mapper/mapper_test.go (hunk 2 above)
Comment (karsov) [reply mapper/mapper_test.go]: There is no "panic" in the code, not sure what you looked at.

**Comment by karsov** [comment]
> I would squash some commits, e.g. the ones that fix the tests/add new tests.

Squashed just one, the other are intentionally left this way to display the logic behind.

File: mapper/mapper_test.go (hunk 1 above)
Comment (mchompalova) [reply mapper/mapper_test.go]: Sorry, I was looking at an outdated commit.

File: _ft/ersatz-fixtures.yml [hunk 4]
```diff
@@ -27,12 +27,28 @@ fixtures:
         annotations:
```
Comment (MiroslavGatsanoga) [inline _ft/ersatz-fixtures.yml]: Should we add fixture for the new endpoint?

File: main.go [hunk 5]
```diff
@@ -30,76 +30,82 @@ func main() {
...
 	})
+	logLevel := app.String(cli.StringOpt{
+		Name:   "log-level",
+		Value:  "INFO",
+		Desc:   "Log level",
+		EnvVar: "LOG_LEVEL",
+	})
 
 	log.SetFormatter(&log.JSONFormatter{})
 	log.SetLevel(log.InfoLevel)
```
Comment (MiroslavGatsanoga) [inline main.go]: Setting the log level here is not needed now.

File: _ft/ersatz-fixtures.yml (hunk 4 above)
Comment (karsov) [reply _ft/ersatz-fixtures.yml]: No, the idea of the fixtures is to provide endpoints for the dependencies of our service.
That's why I've added new requests for getting from public annotations API and the PUT request for the content UUID from the Swagger file.

**Comment by karsov** [comment]
As Marina noticed I had somehow overwritten the HTTP status codes for invalid content/concept UUID to be 500 instead of 400. Now that is fixed.

**Comment by karsov** [comment]
> Still 400 in api.yml

That is correct, the error was in handler/handler.go that I at some step made it return 500 instead of the correct 400 (Bad Request).

**Pull request merged**